### DISCUSSION TITLE
Placas con soporte de multiples puertos midi

### DISCUSSION THREAD

**Discussion by Veroledez** [tttapa/Control-Surface#696] [Q&A]
Hola, gracias por tu trabajo.
Bien, este post no es un problema de la biblioteca, es una duda respecto a que otras placas existentes soportan múltiples puertos midi, se que las teensy soportan hasta 16 puertos, y stm32 también tiene esto, pero no tengo ni idea de como funciona,  entonces me preguntaba que otras placas arruinó o compatibles con arruinó soportan los puertos mido virtuales.

Posdata: las placas teensy en mi ciudad son muy costosas.

Nuevamente muchas gracias por compartir tu trabajo

**Comment by tttapa** [comment]
To get multiple virtual MIDI cables, you need a board that supports multiple USB endpoints. For example, the Arduino Leonardo doesn't have a lot of memory, so defining multiple cables/endpoint might not be possible.  
Most modern ARM-based boards have plenty of memory, and there it's just a matter of actually defining the descriptors for these endpoints. You can look at the Teensy Core as an example, or https://github.com/tttapa/Control-Surface/blob/master/src/MIDI_Interfaces/USBMIDI/mbed/PluggableUSBMIDI-descr.cpp or https://github.com/arduino-libraries/MIDIUSB/blob/master/src/MIDIUSB.cpp.  
You'll have to look at the relevant MIDI and USB standards documents for the full details.
